File: LeetCode/Climbing Stairs.cpp
class Solution {
public:
    typedef long long int ll;
    ll dp[46];
    ll climbStairs(ll n) 
    {
         if(n<3)
         {
             return n;
         }
         if(dp[n]!=0)
         {
             return dp[n];
         }
        dp[n] = climbStairs(n-1)+climbStairs(n-2);
        return dp[n];
    }
};

File: Coding Platforms(codechef,codeforces,etc)/LeetCode/Climbing stairs.java
class Solution {
    public int climbStairs(int n) {
        if (n <= 1) {
            return 1;
        }
        
        int[] ways = new int[n + 1];
        ways[0] = 1;
        ways[1] = 1;
        
        for (int i = 2; i <= n; i++) {
            ways[i] = ways[i - 2] + ways[i - 1];
        }
        
        return ways[n];
    }
}
